core/arch/instr.h

```diff
@@ -1788,6 +1788,17 @@ DR_API
 bool
 instr_writes_to_exact_reg(instr_t *instr, reg_id_t reg, dr_opnd_query_flags_t flags);
 
+DR_API
+/**
+ * Assumes that \p reg is a DR_REG_ constant.
+ * Returns true iff at least one of \p instr's source operands is
+ * the same register (not enough to just overlap) as \p reg.
```

Comment:
This wording sounds like it will exclude a memory operand whose base or index is the exact register which is not what the implementation does.

Reply:
Done